pkg/watch/key_watcher_unix.go

```diff
@@ -0,0 +1,21 @@
+//go:build aix || darwin || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris zos
+
+package watch
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+func enableCharInput(fd int) error {
+	termios, err := unix.IoctlGetTermios(fd, ioctlReadTermios)
+	if err != nil {
+		return err
+	}
+	termios.Lflag &^= unix.ICANON
+	if err := unix.IoctlSetTermios(fd, ioctlWriteTermios, termios); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
I find this to be magic. Can you explain in code comment as to what this is happening here?

Reply:
I've put links on the PRs for you to understand what it is about:

https://man7.org/linux/man-pages/man3/termios.3.html

This is not magic, but terminal settings.